experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,233 @@
+    SpaceGerm{BaseRingType, RingType, SpecType}
+A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct SpaceGerm{BaseRingType<:Ring, RingType<:Ring, SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+
+  function SpaceGerm(X::GermAtClosedPoint)
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X)
+  end
+  
+  function SpaceGerm(X::GermAtGeometricPoint)
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X)
+  end
+end
+
+### Getter functions
+
+function underlying_scheme(X::SpaceGerm)
+  return X.X
+end
+
+@attr Spec function representative(X::SpaceGerm{<:Ring, <:MPolyQuoLocalizedRing})
+    R = base_ring(OO(X))
+    I = modulus(OO(X))
+    Q, _ = quo(R, I)
+    return Spec(Q)
+end
+
+@attr Spec function representative(X::SpaceGerm{<:Ring, <:MPolyLocalizedRing})
+    R = base_ring(OO(X))
+    return Spec(R)
+end
+
+function point(X::SpaceGerm{<:Any, <:Any, <:GermAtClosedPoint})
+  return point_coordinates(inverted_set(OO(X)))
+end
+
+function point(X::SpaceGerm{<:Any, <:Any, <:GermAtGeometricPoint})
+  return prime_ideal(inverted_set(OO(X)))
+end
+
+ring_of_germ(X::AbsSpaceGerm) = OO(X)
+
+function ideal_of_germ(X::AbsSpaceGerm{<:Ring,<:MPolyQuoLocalizedRing})
+    return localized_ring(OO(X))(modulus(OO(X)))
+end
+
+function ideal_of_germ(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return ideal(OO(X),[zero(OO(X))])
+end
+
+function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyQuoLocalizedRing})
+    Y,_ = germ_at_point(localized_ring(OO(X)))
+    return Y
+end
+
+function ambient_germ(X::AbsSpaceGerm{<:Ring,<:MPolyLocalizedRing})
+    return X
+end
+
+############################################################################################################
+# allow user to specify point also as ideal
+# currently broken -- suggestions welcome on how to do this Oscar-style
+############################################################################################################
+function _maxideal_to_point(I::MPolyIdeal)
+  G = groebner_assure(I)
+  dim(I)==0 || error("Ideal does not describe finite set of points")
+  singular_assure(G)
+  vdim(G)==1 || error("Ideal does not describe a single K-point")
+  return [singular.reduce(v,G) for v in gens(base_ring(I))]
+end
+
+### constructors
+function SpaceGerm(X::Spec, a::Vector)
+  R = ambient_ring(X)
+  kk = coefficient_ring(R)
+  b = [kk.(v) for v in a]  ## throws and error, if vector entries are not compatible
+  U = MPolyComplementOfKPointIdeal(R,b)
+  Y = Spec(Localization(OO(X), U)[1])
+  return SpaceGerm(Y)
+end
+  
+function SpaceGerm(X::Spec, I::MPolyLocalizedIdeal)
+  R = base_ring(base_ring(I))
+  R == ambient_ring(X) || error("rings are not compatible")
+  J = ideal(R, [numerator(p) for p in gens(I)]) # expected to be a single k-point, only tested in nex line
```

Comment:
Careful! The generators could very well involve `x*(x-1)` with `(x-1)` a unit in the localization already made. You don't want to catch these. Use `saturated_ideal` instead? Or write another method for `_maxideal_to_point` for localized ideals, using some normal forms in the localized rings.

Reply:
I am aware of that, but since _maxideal_to_point did not even work for a polynomial ring, I did not continue with the local case at all -- in the hope to learn (throught this pull request) how to do _maxideal_to_point properly Oscar-style.